tests/CSharpTemp/CSharpTemp.Tests.cs

```diff
@@ -340,4 +340,35 @@ public void TestNullAttributedFunctionPtr()
         }
     }
 
+    [Test]
+    public void TestMultiOverLoadNPtrToRefTypeGen()
+    {
+        unsafe
+        {
+            var obj = new MultiOverLoadNPtrToRefTypeGenTest();
+            var p = obj.ReturnPrimTypePtr();
+            Assert.AreEqual(0, p[0]);                 // Pass!
+            Assert.AreEqual(0, p[1]);                 // Pass!
+            Assert.AreEqual(0, p[2]);                 // Pass!
+
+            obj.TakePrimTypePtr(ref *p);
+            Assert.AreEqual(100, p[0]);               // Pass!
+            //Assert.AreEqual(200, p[1]);               // Fail!
+            //Assert.AreEqual(300, p[2]);               // Fail!
+            
+            int[] array = { 1, 2, 3 };
+            fixed (int* p1 = array)
+            {
+                obj.TakePrimTypePtr(ref *p1);
+                Assert.AreEqual(100, p1[0]);          //Pass!
+                //Assert.AreEqual(200, p1[1]);          //Fail!
+                //Assert.AreEqual(300, p1[2]);          //Fail!
+            }
+
+            Assert.AreEqual(100, array[0]);           //Pass!!!
+            //Assert.AreEqual(200, array[1]);           //FAIL!!!
+            //Assert.AreEqual(300, array[2]);           //FAIL!!!
+        }
```

Comment:
int _ptr = ptr;

This makes a copy so you pass to C++ the address of _ptr and not of ptr. Could you change that to use ptr directly:
{
    var arg0 = &ptr;
    Internal.TakePrimTypePtr_0(__Instance, arg0);
}

There's a chance this will also fix your tests. As a beginning you could try changing the code manually (you can disable the pre-build event which regenerates the code) and see if it works.